### ISSUE TITLE
[RFC] - Format Sum types

### ISSUE THREAD

**Issue by mauriciofierrom** [codedownio/aeson-typescript#19]
Allow formatting sum types in different fashions.

```haskell
data CountryCode = EC | US | MU deriving (Eq, Show)
```

## String Literal Types

The default and the way it currently works

```typescript
type CountryCode = "EC" | "US" | "MU";
```

## Enum

Create a plain TS enum:

```typescript
enum CountryCode {
    EC,
    US,
    MU,
}
```
## EnumWithType

Creates an enum with a type declaration:

```typescript
enum CountryCodeEnum {
    EC="EC",
    US="US",
    MU="MU"
}

type CountryCode = keyof typeof CountryCodeEnum;
```

This allows us to generate the types from the `Enum`, which is the intended use in TS, and also allows us to generate collections from the `Enum` using `Object.keys(CountryCodeEnum)`.

**Comment by mauriciofierrom** [comment]
If this functionality would be desired I think I have an approach ready for review I can put in a PR.

**Comment by thomasjm** [comment]
Hi @mauriciofierrom, thanks for the RFC. It does sound nice to be able to get the set of possible values from the enum. A PR sounds good to me if you'd like to send it.

(Curious how/if you're planning for this to be configured. It seems like if enums work in all versions of TS then it would be fine to switch completely to the `EnumWithType` version. At the code level, I guess you need to add a new constructor to `TSDeclaration`?)

**Comment by mauriciofierrom** [comment]
I'm adding a `SumTypeFormat` type for the three types of formatting I described earlier and add it as a field for `FormattingOptions`. Then I just added checks in the `formatTSDeclaration` and `formatTSDeclarations'` functions. I've put up the PR so you can check it #20

**Comment by thomasjm** [comment]
Merged in 97d116663a518bd593131f9cbc36697173135768
